db/keyval/etcd/plugin_impl_etcd.go

```diff
@@ -213,34 +217,28 @@ func (plugin *Plugin) configureConnection() {
 	plugin.protoWrapper = kvproto.NewProtoWrapperWithSerializer(plugin.connection, &keyval.SerializerJSON{})
 }
 
-// Starts periodic ETCD status check
-func (plugin *Plugin) startStatusCheck() {
-	// Register for providing status reports (polling mode).
-	if plugin.StatusCheck != nil {
-		plugin.StatusCheck.Register(core.PluginName(plugin.PluginName), func() (statuscheck.PluginState, error) {
-			if plugin.connection == nil {
-				plugin.connected = false
-				return statuscheck.Error, fmt.Errorf("no ETCD connection available")
-			}
-			if _, _, _, err := plugin.connection.GetValue(healthCheckProbeKey); err != nil {
-				plugin.lastConnErr = err
-				plugin.connected = false
-				return statuscheck.Error, err
-			}
-			if plugin.reconnectResync && plugin.lastConnErr != nil {
-				plugin.Log.Info("Starting resync after ETCD reconnect")
-				if plugin.Resync != nil {
-					plugin.Resync.DoResync()
-					plugin.lastConnErr = nil
-				} else {
-					plugin.Log.Warn("Expected resync after ETCD reconnect could not start beacuse of missing Resync plugin")
-				}
+// ETCD status check probe function
+func (plugin *Plugin) statusCheckProbe() statuscheck.PluginStateProbe {
+	return func() (statuscheck.PluginState, error) {
```

Comment:
What's the reason to return function here? You can just define `statusCheckProbe` to have this signature and use `plugin.StatusCheck.Register(plugin.PluginName, plugin.statusCheckProbe)` instead of `plugin.StatusCheck.Register(plugin.PluginName, plugin.statusCheckProbe())` to register it.